Enable pass attributes to script and link

wszgxa opened the pull request:
## Motivation
Github issue: #848 

Enable pass attributes to script tag and link tag.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

I have tested this locally, it will apply the attributes to script and link.

## Related PRs

nope

yangshun commented:
Retried Netlify deploy!

endiliey commented on lib/core/Head.js:
```diff
@@ -134,13 +134,23 @@ class Head extends React.Component {
-          ))}
+          this.props.config.scripts.map(source => {
+            if (source instanceof Object) {
+              return source.src ? (
```
Use [Inline If with Logical && Operator](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)

e.g:
```jsx
return source.src && <script type="text/javascript" key={source.src} {...source} />
```

yangshun commented on docs/api-site-config.md:
```diff
@@ -155,15 +155,15 @@ h1 {
 * `separate` - The secondary navigation is a separate pane defaulting on the right side of a document. See http://docusaurus.io/docs/en/translation.html for an example.
 
-`scripts` - Array of JavaScript sources to load. The script tag will be inserted in the HTML head.
+`scripts` - Array of JavaScript sources to load. You can pass only src link as srting, or you can pass attributes as object. The script tag will be inserted in the HTML head.
```
Typo in `srting`. Use this instead:

```
Array of JavaScript sources to load. The values can be either strings or plain objects of attribute-value maps. Refer to the example below. The script tag will be inserted in the HTML head.
```

yangshun commented on docs/api-site-config.md:
```diff
@@ -155,15 +155,15 @@ h1 {
 `scrollToTopOptions` - Optional options configuration for the scroll to top button. You do not need to use this, even if you set `scrollToTop` to `true`; it just provides you more configuration control of the button. You can find more options [here](https://github.com/vfeskov/vanilla-back-to-top/blob/v7.1.14/OPTIONS.md). By default, we set the zIndex option to 100.
 
-`stylesheets` - Array of CSS sources to load. The link tag will be inserted in the HTML head.
+`stylesheets` - Array of CSS sources to load. You can pass only href link as srting, or you can pass attributes as object. The link tag will be inserted in the HTML head.
```
Similar changes.

yangshun commented on lib/core/Head.js:
```diff
@@ -134,13 +134,23 @@ class Head extends React.Component {
-          this.props.config.scripts.map(source => (
-            <script type="text/javascript" key={source} src={source} />
-          ))}
+          this.props.config.scripts.map(source => {
```
Better written as:

```jsx
this.props.config.scripts.map(script => {
  const scriptProps = Object.assign({}, 
    { type: 'text/javasript' }, 
    script instanceOf 'Object' ? script : { src: source },
  );
  return <script {...source} />;
});
```

wszgxa replied:
We need to include the key, 
``` js
this.props.config.scripts && 
  this.props.config.scripts.map(source => {
    const scriptProps = Object.assign(
        {type: 'text/javasript'},
        source.src ? source : {src: source} // include key in this line is more complexity than use ?..: ...
    );
    return <script {...scriptProps} />;
})
```

I guess if we don't need key, we can do as your example.

yangshun commented on lib/core/Head.js:
```diff
@@ -134,13 +134,23 @@ class Head extends React.Component {
-          this.props.config.stylesheets.map(source => (
-            <link rel="stylesheet" key={source} href={source} />
-          ))}
+          this.props.config.stylesheets.map(source => {
```
Rewrote here in a similar fashion as well.

wszgxa commented:
Will address comments at tonight.

wszgxa commented:
Hi, guys

I have addressed the comments. Because we need to pass down a unique key. 

I guess use ternary operator is more readable than use Object.assign.
